src/view-compiler.js

```diff
@@ -158,7 +158,7 @@ export class ViewCompiler {
       return this._compileElement(node, resources, instructions, parentNode, parentInjectorId, targetLightDOM);
     case 3: //text node
       //use wholeText to retrieve the textContent of all adjacent text nodes.
-      let expression = resources.getBindingLanguage(this.bindingLanguage).inspectTextContent(resources, node.wholeText);
+      let expression = resources.getBindingLanguage(this.bindingLanguage).inspectTextContent(resources, node.wholeText || node.data);
```

Comment:
I'm working on the PR now, there's a problem with the test for it which I should have fixed in a few mins. We definitely could monkey-patch jsdom. I don't think there's a perf issue with my approach, I've added some comments above to try and clarify what I meant.